File: src/main/java/me/lutto/questuhc/manager/ArenaManager.java
package me.lutto.questuhc.manager;

import me.lutto.questuhc.instance.Arena;
import me.lutto.questuhc.QuestUHC;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.List;

public class ArenaManager {

    private List<Arena> arenas = new ArrayList<>();

    public ArenaManager(QuestUHC questUHC) {

        FileConfiguration config = questUHC.getConfig();
        for (String arenaConfig : config.getConfigurationSection("arenas.").getKeys(false)) {
            arenas.add(new Arena(questUHC, Integer.parseInt(arenaConfig),
                    new Location(Bukkit.getWorld(config.getString("arenas." + arenaConfig + ".world")),
                            config.getDouble("arenas." + arenaConfig + ".spawn-location.x"),
                            config.getDouble("arenas." + arenaConfig + ".spawn-location.y"),
                            config.getDouble("arenas." + arenaConfig + ".spawn-location.z"),
                            (float) config.getDouble("arenas." + arenaConfig + ".spawn-location.yaw"),
                            (float) config.getDouble("arenas." + arenaConfig + ".spawn-location.pitch")),
                    new Location(Bukkit.getWorld(config.getString("arenas." + arenaConfig + ".world")),
                            config.getDouble("arenas." + arenaConfig + ".first-corner.x"),
                            config.getDouble("arenas." + arenaConfig + ".first-corner.y"),
                            config.getDouble("arenas." + arenaConfig + ".first-corner.z")),
                    new Location(Bukkit.getWorld(config.getString("arenas." + arenaConfig + ".world")),
                            config.getDouble("arenas." + arenaConfig + ".second-corner.x"),
                            config.getDouble("arenas." + arenaConfig + ".second-corner.y"),
                            config.getDouble("arenas." + arenaConfig + ".second-corner.z"))));
        }

    }

    public List<Arena> getArenas() { return arenas; }

    public Arena getArena(Player player) {

        for (Arena arena : arenas) {
            if (arena.getPlayers().contains(player.getUniqueId())) {
                return arena;
            }
        }

        return null;

    };

    public Arena getArena(int id) {

        for (Arena arena : arenas) {
            if (arena.getId() == id) {
                return arena;
            }
        }

        return null;

    };

}

File: src/main/kotlin/me/lutto/questuhc/manager/ArenaManager.kt
package me.lutto.questuhc.manager

import me.lutto.questuhc.QuestUHC
import me.lutto.questuhc.instance.Arena
import org.bukkit.Bukkit
import org.bukkit.Location
import org.bukkit.entity.Player

class ArenaManager(questUHC: QuestUHC) {

    private val arenas: MutableList<Arena> = ArrayList()

    init {
        val config = questUHC.config
        for (arenaConfig in config.getConfigurationSection("arenas.")!!.getKeys(false)) {
            arenas.add(Arena(
                    questUHC, arenaConfig.toInt(),
                    Location(
                        Bukkit.getWorld(config.getString("arenas.$arenaConfig.world")!!),
                        config.getDouble("arenas.$arenaConfig.spawn-location.x"),
                        config.getDouble("arenas.$arenaConfig.spawn-location.y"),
                        config.getDouble("arenas.$arenaConfig.spawn-location.z"),
                        config.getDouble("arenas.$arenaConfig.spawn-location.yaw").toFloat(),
                        config.getDouble("arenas.$arenaConfig.spawn-location.pitch").toFloat()
                    ),
                    Location(
                        Bukkit.getWorld(config.getString("arenas.$arenaConfig.world")!!),
                        config.getDouble("arenas.$arenaConfig.first-corner.x"),
                        config.getDouble("arenas.$arenaConfig.first-corner.y"),
                        config.getDouble("arenas.$arenaConfig.first-corner.z")
                    ),
                    Location(
                        Bukkit.getWorld(config.getString("arenas.$arenaConfig.world")!!),
                        config.getDouble("arenas.$arenaConfig.second-corner.x"),
                        config.getDouble("arenas.$arenaConfig.second-corner.y"),
                        config.getDouble("arenas.$arenaConfig.second-corner.z")
                    )
            ))
        }
    }

    fun getArenas(): List<Arena> = arenas

    fun getArena(player: Player): Arena? {
        for (arena in arenas) {
            if (arena.getPlayers().contains(player.uniqueId)) {
                return arena
            }
        }

        return null
    }

    fun getArena(id: Int): Arena? {
        for (arena in arenas) {
            if (arena.getId() == id) {
                return arena
            }
        }

        return null
    }

}
